baseapp/state.go

```diff
@@ -1,14 +1,18 @@
 package baseapp
 
 import (
+	"sync"
+
 	storetypes "cosmossdk.io/store/types"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 type state struct {
-	ms  storetypes.CacheMultiStore
+	mtx sync.RWMutex
 	ctx sdk.Context
+	// no need for synchronization on the MultiStore as implementations are thread-safe
+	ms storetypes.CacheMultiStore
```

Comment:
Generally, you put the mtx above all the fields it synchronizes. This is how the reader knows ;-) 

```suggestion
	ms storetypes.CacheMultiStore
	
	mtx sync.RWMutex
	ctx sdk.Context
```